ddtrace/_trace/processor/__init__.py

```diff
@@ -324,6 +324,14 @@ def on_span_finish(self, span):
 
                 if should_partial_flush:
                     log.debug("Partially flushing %d spans for trace %d", num_finished, span.trace_id)
+                    if not finished:
+                        log_msg = f"No spans to flush in partial flush for span: {span} (was tracer reconfigured?)"
+                        # Avoid potential crashes if, for some reason, we have no spans to flush even though
+                        # should_partial_flush is True.
+                        if config._telemetry_enabled:
+                            telemetry.telemetry_writer.add_log("WARNING", log_msg)
+                        log.warning(log_msg)
+                        return
```

Comment:
thinking about this, do we need to not increment `num_finished` in this case as well?

which makes me wonder if instead we need something like:

```python
if span not in trace:
    log.warning('wtf?')
    return
```

Reply:
I think you're right. I didn't realize `trace.num_finished` was just an attribute we increment/decrement "manually".

Reply:
I think that makes the original change under `if should_partial_flush:` more or less unreachable but "you never know".

Side note, some tests output:
```
tests/internal/test_settings.py::test_tracer_reconfigure_does_not_crash_tracer
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/tests/internal/test_settings.py(624)test_tracer_reconfigure_does_not_crash_tracer()
-> assert status == 0, f"err={err.decode('utf-8')} out={out.decode('utf-8')}"
(Pdb) err
b'No trace found for span <Span(id=3488788091398017934,trace_id=135992766012335207893023319893039562152,parent_id=None,name=regression)> (was tracer reconfigured?)\n'
```

```
tests/internal/test_settings.py::test_tracer_trace_removed_does_not_crash
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/tests/internal/test_settings.py(642)test_tracer_trace_removed_does_not_crash()
-> assert status == 0, f"err={err.decode('utf-8')} out={out.decode('utf-8')}"
(Pdb) err
b'No trace found for span <Span(id=9329823571518464311,trace_id=135992766250019695449858101370167084146,parent_id=4517369765126936596,name=regression)> (was tracer reconfigured?)\n'
```

```
tests/internal/test_settings.py::test_tracer_trace_span_removed_does_not_crash
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB set_trace >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /Users/romain.komorn/go/src/github.com/DataDog/dd-trace-py/tests/internal/test_settings.py(660)test_tracer_trace_span_removed_does_not_crash()
-> assert status == 0, f"err={err.decode('utf-8')} out={out.decode('utf-8')}"
(Pdb) err
b'Finished span <Span(id=3480700046215677949,trace_id=135992766170791532931548392515882328551,parent_id=11076758516579201008,name=regression)> not found in trace 135992766170791532931548392515882328551\n'
```